### PR TITLE
Throw on label redefinition in interpreter.

### LINKED ISSUE #7901: Interpreter fails to throw when compiling redefined labels.
A label re-defined in the same scope should throw upon compilation, but this does not happen in the interpreter.

This bug was introduced by me at https://github.com/dotnet/corefx/pull/6993/files#diff-e009724c302721a6062e8980279534feR71

### PR DESCRIPTION
**Pull request by JonHanna**
Fixes #7901

Also includes quite full set of tests for loop expressions, which uncovered this.

@stephentoub, @VSadov PTAL.

### CODE REVIEW

**Comment by JonHanna** [comment]
Test Innerloop OSX Debug Build and Test
Test Innerloop OSX Release Build and Test
(Failed to launch)

**Comment by mmitche** [comment]
Test Innerloop OSX Debug Build and Test
Test Innerloop OSX Release Build and Test

**Comment by JonHanna** [comment]
Test Innerloop OSX Debug Build and Test
Test Innerloop OSX Release Build and Test

File: src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LabelInfo.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ internal void Define(LabelScopeInfo block)
             {
                 if (j.ContainsTarget(_node))
                 {
-                    Error.LabelTargetAlreadyDefined(_node.Name);
```
Comment (VSadov) [inline src/System.Linq.Expressions/src/System/Linq/Expressions/Interpreter/LabelInfo.cs]: clearly "throw" is mising.

**Comment by VSadov** [comment]
LGTM, Thanks!

**Pull request merged**